bots/activity.bot.js

```diff
@@ -0,0 +1,84 @@
+const _ = require('lodash');
+const config = require('pico-conf');
+const Slack = require('pico-slack');
+
+
+const DEBUG = config.get('activitybot.debug');
+const TARGET_CHANNEL = config.get('activitybot.target_channel');
+
+const MESSAGE_COUNT_THRESHOLD = parseInt(
+	config.get('activitybot.threshold.message_count')
+);
+if (isNaN(MESSAGE_COUNT_THRESHOLD)) {
+	Slack.error(
+		'[ActivityBot]: Could not parse message count threshold config value'
+	);
+	return;
+}
+
+const THRESHOLD_SECONDS = parseInt(
+	config.get('activitybot.threshold.seconds')
+);
+if (isNaN(THRESHOLD_SECONDS)) {
+	Slack.error('[ActivityBot]: Could not parse time threshold config value');
+	return;
+}
+
+const IGNORED_CHANNELS = config.get('activitybot.ignored_channels').split(',')
+
+
+const messageTimestampsByChannel = {};
+
+
+/**
+ * Add `msg`'s timestamp as UNIX epoch to the end of the array for
+ * the message's channel, initializing the array if necessary.
+ * The timestamps are appended, meaning that the array should be sorted,
+ * more or less.
+ */
+const tallyMessage = (msg) => {
+	if (_.in(msg.channel, IGNORED_CHANNELS)) {
+		return;
+	}
+	if (!_.has(messageTimestampsByChannel, msg.channel)) {
+		messageTimestampsByChannel[msg.channel] = [];
+	}
+	const unixTimestamp = Math.floor(msg.ts * 1000);
+	messageTimestampsByChannel[msg.channel].push(unixTimestamp)
```

Comment:
It doesn't look like there's anything stopping this array from growing infinitely?  Aside from dyno cycling.

Reply:
The `checkForActivityBursts` function updates the array with the culled version, so these should stay quite short.